### PR TITLE
Add ZERODRAG WARPF7

### PR DESCRIPTION
**Pull request by haslinghuis**

### CODE REVIEW

**Review by nerdCopter — APPROVED** [review]

File: configs/WARPF7/config.h [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+#define MAG_I2C_INSTANCE (I2CDEV_1)
+#define DEFAULT_BLACKBOX_DEVICE BLACKBOX_DEVICE_FLASH
+#define DEFAULT_CURRENT_METER_SOURCE CURRENT_METER_ADC
+#define DEFAULT_VOLTAGE_METER_SOURCE VOLTAGE_METER_ADC
+#define PINIO1_BOX 40
+#define PINIO2_BOX 41
+
+#define SERIALRX_UART SERIAL_PORT_USART1
+#define ESC_SENSOR_UART SERIAL_PORT_USART2
+#ifdef USE_OSD_SD
```
Comment (ot0tot) [inline configs/WARPF7/config.h:138]: What happens if both `USE_OSD_SD` and `USE_OSD_HD` are defined? 

Should the GPS UART use a `#ifdef GPS` also?

File: configs/WARPF7/config.h [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    TIMER_PIN_MAP( 0, MOTOR1_PIN, 2, 0) \
+    TIMER_PIN_MAP( 1, MOTOR2_PIN, 2, 0) \
+    TIMER_PIN_MAP( 2, MOTOR3_PIN, 1, 0) \
+    TIMER_PIN_MAP( 3, MOTOR4_PIN, 1, 0) \
+    TIMER_PIN_MAP( 4, MOTOR5_PIN, 2, 0) \
+    TIMER_PIN_MAP( 5, MOTOR6_PIN, 2, 0) \
+    TIMER_PIN_MAP( 6, MOTOR7_PIN, 2, 0) \
+    TIMER_PIN_MAP( 7, MOTOR8_PIN, 2, 0) \
+    TIMER_PIN_MAP( 8, LED_STRIP_PIN, 1, 0) \
+    TIMER_PIN_MAP( 9, CAMERA_CONTROL_PIN, 1, 0) \
```
Comment (ot0tot) [inline configs/WARPF7/config.h:122]: Camera control and RX_PPM do not need DMA. Does this FC actually have a PPM input pad?

Comment (haslinghuis) [reply configs/WARPF7/config.h]: PA1 was on the schematics but can't find it on the wire diagram.

Thanks for review. Fixed.

File: configs/WARPF7/config.h (hunk 1 above)
Comment (haslinghuis) [reply configs/WARPF7/config.h:135]: Agree `USE_OSD_SD` and `USE_OSD_HD` should be exclusive - when flashing firmware.

For GPS need to review when sample arrives.

**Comment by haslinghuis** [comment]
```bash
# status
MCU F722 Clock=216MHz, Vref=3.29V, Core temp=52degC
Stack size: 2048, Stack address: 0x20010000
Configuration: CONFIGURED, size: 3670, max available: 16384
Devices detected: SPI:2, I2C:0
Gyros detected: gyro 1 locked dma
GYRO=ICM42688P, ACC=ICM42688P, BARO=BMP280
OSD: MAX7456 (30 x 13)
System Uptime: 396 seconds, Current Time: 2024-07-01T17:01:10.806+00:00
CPU:31%, cycle time: 124, GYRO rate: 8064, RX rate: 15, System rate: 9
Voltage: 0 * 0.01V (0S battery - NOT PRESENT)
I2C Errors: 0
FLASH: JEDEC ID=0x00ef4019 32M
GPS: connected, UART4 57600 (set to 57600), configured, version =  M10
Arming disable flags: RXLOSS CLI MSP
```

```bash
# tasks
Task list             rate/hz  max/us  avg/us maxload avgload  total/ms   late    run reqd/us
00 - (         SYSTEM)     10       1       0    0.0%    0.0%         1      0     12       0
01 - (         SYSTEM)    989       4       0    0.3%    0.0%       180      0   1159       3
02 - (           GYRO)   8078       4       1    3.2%    1.2%      4145      0   9444       0
03 - (         FILTER)   8078      15       9   12.1%    7.7%     26059      0   9444       0
04 - (            PID)   8078      37      25   29.8%   20.6%     68041      0   9444       0
05 - (            ACC)    991       7       2    0.6%    0.2%       818      0   1158       5
06 - (       ATTITUDE)    100      10       8    0.1%    0.0%       285      0    117      11
07 - (             RX)     10      26      17    0.0%    0.0%       167      0     36      27
08 - (         SERIAL)    100    4488       2   44.8%    0.0%       878      0    116      15
09 - (       DISPATCH)    991       5       0    0.4%    0.0%       265      1   1159       3
10 - (BATTERY_VOLTAGE)     50       3       2    0.0%    0.0%        32      0     59       2
11 - (BATTERY_CURRENT)     50       2       0    0.0%    0.0%        10      0     58       1
12 - ( BATTERY_ALERTS)      5       5       2    0.0%    0.0%         4      0      6       4
13 - (         BEEPER)    100       2       0    0.0%    0.0%        27      0    116       1
14 - (            GPS)    224      34       9    0.7%    0.2%       555      0    200      33
15 - (     GPS_RESCUE)    100       2       1    0.0%    0.0%        43      0    117       2
16 - (           BARO)     40      23      16    0.0%    0.0%       641      0    141      26
17 - (       ALTITUDE)    100       4       2    0.0%    0.0%        83      0    116       3
19 - (            OSD)     12      34      17    0.0%    0.0%       996      0    259      27
21 - (            CMS)     20       2       1    0.0%    0.0%        12      0     24       5
22 - (        VTXCTRL)      5       3       1    0.0%    0.0%        28      0      6      31
23 - (        CAMCTRL)      5       1       0    0.0%    0.0%         1      0      6       3
25 - (    ADCINTERNAL)      1       4       3    0.0%    0.0%         1      0      1       7
27 - (SPEED_NEGOTIATION)    100      10       1    0.1%    0.0%        54      0    116       9
RX Check Function                   8       2                       229
Total (excluding SERIAL)                                29.9%
```

```bash
resource show all
# resource show all
Currently active IO resource assignments:
(reboot to update)
--------------------
A00: SERIAL_TX 4
A01: SERIAL_RX 4
A04: GYRO_CS 1
A05: SPI_SCK 1
A06: SPI_SDI 1
A07: SPI_SDO 1
A09: SERIAL_TX 1
A10: SERIAL_RX 1
A11: USB
A12: USB
A13: SWD
A14: SWD
A15: BEEPER
B00: MOTOR 1
B01: MOTOR 2
B02: LED 2
B03: SPI_SCK 3
B04: SPI_SDI 3
B05: SPI_SDO 3
B06: FLASH_CS
B07: CAMERA_CONTROL
B08: I2C_SCL 1
B09: I2C_SDA 1
B10: MOTOR 3
B11: MOTOR 4
B12: OSD_CS
B13: SPI_SCK 2
B14: SPI_SDI 2
B15: SPI_SDO 2
C01: ADC_CURR
C02: ADC_BATT
C03: LED 1
C04: GYRO_EXTI
C10: SERIAL_TX 3
C13: BARO_CS
C14: PINIO 1
C15: PINIO 2

Currently active Timers:
-----------------------
TIM1: FREE
TIM2: FREE
TIM3: FREE
TIM4:
    CH2 : CAMERA_CONTROL
TIM5: FREE
TIM6: FREE
TIM7: FREE
TIM8:
    CH1 : DSHOT_BITBANG 2
TIM9: FREE
TIM10: FREE
TIM11: FREE
TIM12: FREE
TIM13: FREE
TIM14: FREE

Currently active DMA:
--------------------
DMA1 Stream 0: SPI_SDI 3
DMA1 Stream 1: FREE
DMA1 Stream 2: FREE
DMA1 Stream 3: SPI_SDI 2
DMA1 Stream 4: SPI_SDO 2
DMA1 Stream 5: SPI_SDO 3
DMA1 Stream 6: FREE
DMA1 Stream 7: FREE
DMA2 Stream 0: SPI_SDI 1
DMA2 Stream 1: FREE
DMA2 Stream 2: DSHOT_BITBANG 2
DMA2 Stream 3: SPI_SDO 1
DMA2 Stream 4: ADC
DMA2 Stream 5: FREE
DMA2 Stream 6: FREE
DMA2 Stream 7: FREE
```

**Comment by ot0tot** [comment]
Looks like LED_STRIP is on TIM1, so there shouldn't be any issues with that enabled. Worth verifying though.

Have you noticed any issues with blackbox and baro sharing SPI2? I assume flash will be primarily write and baro primarily read so it will probably be OK. Baro on I2C would be preferable though.

**Review by ot0tot — APPROVED** [review]

**Pull request merged**